src/main/java/dev/lambdaurora/lambdynlights/mixin/sodium/LightDataAccessMixin.java

```diff
@@ -0,0 +1,32 @@
+package dev.lambdaurora.lambdynlights.mixin.sodium;
+
+import dev.lambdaurora.lambdynlights.LambDynLights;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.BlockRenderView;
+import net.minecraft.world.LightType;
+import org.spongepowered.asm.mixin.Dynamic;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Pseudo;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+@Pseudo
+@Mixin(targets = "me.jellysquid.mods.sodium.client.model.light.data.LightDataAccess", remap = false)
+public class LightDataAccessMixin {
+    // In Sodium 0.5, LightDataAccess has replaced the Vanilla LightmapTextureManager,
+    // and no longer uses WorldRenderer#getLightmapCoordinates.
+    // Unfortunately, this breaks LambDynLights, and as a result is forced to mixin to Sodium.
+    @Dynamic
+    @Redirect(method = "compute", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/BlockRenderView;getLightLevel(Lnet/minecraft/world/LightType;Lnet/minecraft/util/math/BlockPos;)I", ordinal = 0, remap = true), remap = false)
+    private int lambdynlights$modifyBlockLight(BlockRenderView world, LightType type, BlockPos pos, int x, int y, int z) {
+        int original = world.getLightLevel(type, pos);
+
+        if (!world.getBlockState(pos).isOpaqueFullCube(world, pos) && LambDynLights.get().config.getDynamicLightsMode().isEnabled()) {
```

Comment:
This `!world.getBlockState(pos).isOpaqueFullCube(world, pos)` check is somewhat inefficient - Sodium already computed it before: https://github.com/CaffeineMC/sodium-fabric/blob/21e4e6d2f09f8d6b8ae7e5a5fab4a13c1f8379ce/src/main/java/me/jellysquid/mods/sodium/client/model/light/data/LightDataAccess.java#L68.

Reply:
i'm aware of this, but i couldn't manage to figure out how to recapture the locals without needing MixinExtra.

Reply:
you can actually see remnants of my feeble attempt of trying to capture locals from the `int x, int y, int z` params that I forgot to remove

Reply:
Ah yes, it seems that redirect cannot capture local variables. Oh well.